shark_turbine/kernel/wave/promotion.py

```diff
@@ -39,12 +39,10 @@ def promote_node(
 
     assert isinstance(node, Read) or isinstance(node, Write)
     with node.graph.inserting_before(node.fx_node.next):
-        workgroup_distributed_shape = get_workgroup_distributed_shape(
-            node.type.symbolic_shape, constraints
-        )
+        constrained_shape = get_constrained_shape(node.type.symbolic_shape, constraints)
```

Comment:
Does this mean we anticipate promoting memory that is not part of the workgroup level tiling? I guess that makes sense for case where we are tiling reduction dims i.e workgroup_K is typically full size but we may tile it within a loop and only want to promote the TilingConstraint or IV size . 

Is that correct, or are there other use cases that need this for?

Reply:
Usually, we would only promote after workgroup distributing and any sort of tiling because we want to promote the smallest shape possible (since shared memory is a limited resource). This is only for that use case.